refactor(oxfmt): Use stdout and stderr appropreately

leaysgur opened the pull request:
Until now, everything was displayed to `stdout`.

This caused inconveniences when combining pipes in the CLI.
For example, the following does not work properly in some situation.

```sh
oxfmt dir_with_parse_error --list-different | wc -l
```

Because the graphical parse error flows along with the paths that has updated.

This PR solves this.

leaysgur commented:
* **#15853** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15853?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15853?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15853).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
### Merge activity

* **Nov 19, 5:45 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 19, 5:45 AM UTC**: `Dunqing` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

leaysgur committed:
refactor(oxfmt): Use stdout and stderr appropreately (#15853)

Until now, everything was displayed to `stdout`.

This caused inconveniences when combining pipes in the CLI.
For example, the following does not work properly in some situation.

```sh
oxfmt dir_with_parse_error --list-different | wc -l
```

Because the graphical parse error flows along with the paths that has updated.

This PR solves this.
